platform-docs/docs/intro.md

```diff
@@ -30,13 +30,39 @@ To build a block editor, you need to install the following dependencies:
 
 We're going to be using JSX to write our UI and components. So one of the first steps we need to do is to configure our build tooling, By default vite supports JSX and and outputs the result as a React pragma. The Block editor uses React so there's no need to configure anything here but if you're using a different bundler/build tool, make sure the JSX transpilation is setup properly.
 
+## Side note: `process.env.IS_GUTENBERG_PLUGIN`
+
+The Gutenberg block editor reads from the `process.env.IS_GUTENBERG_PLUGIN` variable. It
+is used to distinguish the code that is part of the Gutenberg block editor
+[plugin](https://wordpress.org/plugins/gutenberg/) and the code that is part of
+WordPress core. 
+
+For our purposes, as we're building a standalone block editor that is completely
+separate from WordPress, we want to set this variable to `false`. 
+
+When using Vite, we can do it by creating a `vite.config.js` file:
+
+```js
+// vite.config.js
+import { defineConfig } from "vite";
+
+export default defineConfig({
+  define: {
+    "process.env.IS_GUTENBERG_PLUGIN": JSON.stringify(false),
+  },
+});
+```
+
+If you are using Webpack, you can use the
+[DefinePlugin](https://webpack.js.org/plugins/define-plugin/) to accomplish the
+same thing.
+
 ## Bootstrap your block editor
 
-It's time to render our first block editor.
+It's time to render our first block editor. Update your `index.jsx` file with the following code:
 
- - Update your `index.jsx` file with the following code:
 ```jsx
-import { createElement, useState } from "react";
+import React, { useState } from "react";
```

Comment:
This is fine but I kind of thought that vite uses the automatic JSX config by default. it's not the case?

Reply:
There are more details in https://v2.vitejs.dev/guide/features.html#jsx but TL;DR is that it's not the case out of the box. We can also do one of those things

- use the React plugin https://www.npmjs.com/package/@vitejs/plugin-react
- add sth like those lines to the vite config:
   ```js
   export default defineConfig({
    esbuild: {
        jsxInject: `import React from 'react'`
      }
   })
```

Reply:
Let's keep it simple and just import React like you did for now.

Reply:
If you create the Vite app with the "React" / "JavaScript" config instead of "Vanilla" / "JavaScript" config, this works automatically. The vite config becomes:

```js
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
});
```

Reply:
Actually, starting with the "Vanilla" vite config cerates a few issues (jsx, react), why not change the docs to suggeste starting with the "React" vite config? Those things will be solved.

Reply:
If starting with "Vanilla" then `react` and `react-dom` aren’t added as dependencies so it seems like that would warrant mentioning them in "Installing dependencies" section.

I'd favor instructing to use the "React" variant. It seems like it simplifies the instructions and provides a better starting place as it adds some helpful packages like eslint with react-related plugins.